cardinal/system/system_create_player.go

```diff
@@ -73,16 +75,65 @@ func CreatePlayerSystem(world cardinal.WorldContext) error {
 				}, err
 			}
 
-			// Set the Player Entity variables of the created player entity
+			// Retrieve all rooms and select a random one.
+			var roomIDs []types.EntityID
+			err = cardinal.NewSearch().Entity(filter.Exact(filter.Component[component.Room]())).Each(world, func(id types.EntityID) bool {
+				roomIDs = append(roomIDs, id)
+				return true
+			})
+
+			if err != nil || len(roomIDs) == 0 {
+				err := fmt.Errorf("no rooms available to add the player")
+				log.Println(err)
+				return msg.CreatePlayerReply{
+					Success: false,
+					Message: err.Error(),
+				}, err
+			}
+
+			rand.Seed(time.Now().UnixNano())
+			selectedRoomID := roomIDs[rand.Intn(len(roomIDs))]
+
+			playerID := uint32(playerManagerID) // Convert EntityID to uint32
```

Comment:
Ok I can specify on which room they start.  The code is for one player. So when you create 1 player he is added in 1 room. You will have to create another one (through the transaction) to do the same. When you say at the Start what do you mean? Like on Init? If so, then it will no longer be a transaction but rather as the rooms, it will be an init system then

Reply:
player start at  room 1

Reply:
pass the room id to the function

Reply:
Ok, I'll adjust the code tomorrow morning

Reply:
Done, uses the variable that is specified on the message